<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>413</width>
    <height>214</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Low orbit ion cannon control panel</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="openKeyButt">
          <property name="text">
           <string>Open key file</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="genKeyButt">
          <property name="text">
           <string>Generate key file</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="encryptButt">
          <property name="text">
           <string>Encrypt firmware</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Open an existing key file</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Generate new key file to use with bootloader</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Select Intel HEX file to encrypt</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QCheckBox" name="useZeroPageBox">
      <property name="text">
       <string>Use 0th FLASH page</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Page size</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>FLASH size</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QSpinBox" name="pageSizeBox">
        <property name="minimum">
         <number>16</number>
        </property>
        <property name="maximum">
         <number>999999984</number>
        </property>
        <property name="singleStep">
         <number>16</number>
        </property>
        <property name="value">
         <number>128</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="flashSizeBox">
        <property name="minimum">
         <number>16</number>
        </property>
        <property name="maximum">
         <number>999999984</number>
        </property>
        <property name="singleStep">
         <number>16</number>
        </property>
        <property name="value">
         <number>16384</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="uidLabel">
      <property name="text">
       <string>Key UID: N/A</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Based on EAX authenticated encryption scheme
PRNG  used is AutoSeededRandomPool from Crypto++</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
